config/v1alpha1/types_image_policy.go

```diff
@@ -0,0 +1,215 @@
+	// "ExactRepository" means that the identity in the signature must be in the same repository as a specific identity specified by "repository".
+	// "RemapIdentity" means that the signature must be in the same as the remapped image identity. Remapped image identity is obtained by replacing the "prefix" with the specified “signedPrefix” if the the image identity matches the specified remapPrefix.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+	MatchPolicy IdentityMatchPolicy `json:"matchPolicy"`
+	// exactRepository is required if matchPolicy is set to "ExactRepository".
+	// +optional
+	PolicyMatchExactRepository *PolicyMatchExactRepository `json:"exactRepository,omitempty"`
+	// remapIdentity is required if matchPolicy is set to "RemapIdentity".
+	// +optional
+	PolicyMatchRemapIdentity *PolicyMatchRemapIdentity `json:"remapIdentity,omitempty"`
+}
+
+type PolicyMatchExactRepository struct {
+	// repository is the reference of the image identity to be matched.
+	// the value should be a repository (by omitting the tag or digest) in a registry implementing the "Docker Registry HTTP API V2". For example, docker.io/library/busybox
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=512
+	// +kubebuilder:validation:XValidation:rule=`self.matches('^(((?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(?:\\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))+(?::[0-9]+)?)|(localhost(?::[0-9]+)?))(?:(?:/[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?)+)?$')`,message="invalid repository in the signedIdentity"
+	Repository string `json:"repository"`
+}
+
+type PolicyMatchRemapIdentity struct {
+	// prefix is the prefix of the image identity to be matched.
+	// If the image identity matches the specified prefix, that prefix is replaced by the specified “signedPrefix” (otherwise it is used as unchanged and no remapping takes place).
+	// The prefix and signedPrefix values can be either host[:port] values (matching exactly the same host[:port], string), repository namespaces,
+	// or repositories (i.e. they must not contain tags/digests), and match as prefixes of the fully expanded form.
+	// For example, docker.io/library/busybox (not busybox) to specify that single repository, or docker.io/library (not an empty string) to specify the parent namespace of docker.io/library/busybox.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=512
+	// +kubebuilder:validation:XValidation:rule=`self.matches('^((((?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(?:\\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))?(?::[0-9]+)?)|(localhost(?::[0-9]+)?))(?:(?:/[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?)+)?)$')`,message="invalid repository prefix in the signedIdentity"
+	Prefix string `json:"prefix"`
+	// signedPrefix is the prefix of the image identity to be matched in the signature. The format is the same as "prefix".
```

Comment:
How do prefix and signedPrefix relate? They are both required, must prefix and signedPrefix differ?

Reply:
prefix is the prefix of the image that needs to be verified. The image identity used to verify the signature is constructed by replacing the prefix with a signedPrefix. It can work in this way if the prefix and signedprefix are the same, but most users will configure different prefix and signedPrefix.

Reply:
So we are saying that, for example, I had originally got an image and signed it as foo.bar/baz, but then mirrored it, so now it's baz.foo/baz, so the prefix would be baz.foo but the signed prefix would the original, foo.bar?

So this is helpful for use when there's mirroring involved?

Can we expand the comment to capture a bit more of how this field is used and what its purpose is?

Reply:
added the comment https://github.com/openshift/api/pull/1457/files#diff-d21f452e5fe14373caddec257440ba84f6618661d208820eb359ab8c69caae2cR177

Reply:
The same information should be present on this side as well. A user may describe just this field so putting the information on another field wouldn't help them in that case